pkg/sdk/application.go

```diff
@@ -0,0 +1,37 @@
+package sdk
+
+import (
+	"context"
+
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+)
+
+// ApplicationSuppliers is the structure that represents a session's end block height
+// and its matching suppliers.
+type ApplicationSuppliers struct {
+	// Application is the fully hydrated session object returned by the query.
+	Application []apptypes.Application
+}
+
+// GetApplicationSupplierEndpoints returns a flattened structure of the endpoints
+// from all suppliers in the session and returns them as a SupplierEndpoint slice.
+// It queries for the latest session and caches it if the cached one is outdated.
+func (sdk *poktrollSDK) GetApplications(ctx context.Context, gateway string) ([]apptypes.Application, error) {
```

Comment:
WDYT of renaming this to `GetDelegatingApplycationsByGateway` or `GetApplicationsDelegatingToGateway`?

cc @Olshansk

Reply:
That sounds alright with me.

We should decide whether querying all on-chain applications should be a part of an SDK meant to develop gateways that would have no interest in other applications.

For the moment I added a options struct for this method.